src/datasources/blockchain/blockchain-api.service.ts

```diff
@@ -0,0 +1,47 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { Chain as DomainChain } from '@/domain/chains/entities/chain.entity';
+import { RpcUriAuthentication } from '@/domain/chains/entities/rpc-uri-authentication.entity';
+import { IBlockchainApi } from '@/domain/interfaces/blockchain-api.interface';
+import { Injectable } from '@nestjs/common';
+import { Chain, PublicClient, createPublicClient, http } from 'viem';
+
+@Injectable()
+export class BlockchainApi implements IBlockchainApi {
+  private readonly infuraApiKey: string;
+
+  constructor(
+    private readonly configurationService: IConfigurationService,
+    private readonly chain: DomainChain,
+    public readonly destroyClient: (chainId: string) => void,
```

Comment:
What's the purpose of this function? I understand `BlockchainApiManager.destroyBlockchainApi` is intended to delete the `BlockChainApi` entry in the map, but in this class this function doesn't do anything, right? It's intended to release some resources?

Reply:
It is a [remnant from trialling the implementation](https://github.com/safe-global/safe-client-gateway/pull/1623#discussion_r1635019968). I removed it in e38f10916a1a2c4fba22cc228849c2777a96d1c6.